### PR TITLE
Plotting helpers for TiledDataset

### LINKED ISSUE #405: Add a `.plot` helper which produces the tiled plot in https://docs.dkist.nso.edu/projects/python-tools/en/stable/howto_guides/reproject_vbi_mosaic.html

### PR DESCRIPTION
**Pull request by SolarDrew**
Adds a `TiledDataset.plot()` method which creates a quick-look plot as per the example [here](https://docs.dkist.nso.edu/projects/python-tools/en/stable/howto_guides/reproject_vbi_mosaic.html).

At the moment only plots the first image in each tile. Unsure whether to try and construct an animation like the normal dataset plotting or to pass an index argument to the plot to determine which slice is shown.

Closes #405

### CODE REVIEW

**Comment by Cadair** [comment]
@SolarDrew can you rebase this horror show?

File: dkist/dataset/tests/test_tiled_dataset.py [hunk 1]
```diff
@@ -59,3 +61,18 @@ def test_tiled_dataset_from_components(dataset):
         assert ds.files == fm
         assert ds.meta["inventory"] is inventory
         assert ds.meta["headers"] is headers
+
+
+# @pytest.mark.mpl_image_compare
```
Comment (Cadair) [inline dkist/dataset/tests/test_tiled_dataset.py:66]:
```suggestion
```

File: dkist/dataset/tests/test_tiled_dataset.py [hunk 2]
```diff
@@ -59,3 +61,18 @@ def test_tiled_dataset_from_components(dataset):
...
+@figure_test
+@pytest.mark.parametrize("share_scale", [True, False])
+def test_plot(share_scale):
```
Comment (Cadair) [inline dkist/dataset/tests/test_tiled_dataset.py:69]:
```suggestion
@pytest.mark.parametrize("share_zscale", [True, False], ids=["share_zscale", "indpendent_zscale"])
def test_tileddataset_plot(share_zscale):
```

**Review by Cadair — COMMENTED** [review]
Looks like codecov isn't uploading from the figure tests?

File: dkist/dataset/tests/test_tiled_dataset.py (hunk 1 above)
Comment (SolarDrew) [reply dkist/dataset/tests/test_tiled_dataset.py:66]: I thought this was redundant with `@figure_test`?

Comment (SolarDrew) [reply dkist/dataset/tests/test_tiled_dataset.py:66]: Never mind I misread the suggestion. As you were.

**Comment by SolarDrew** [comment]
I think this is good to go unless there are any more issues you can see with it.

File: dkist/dataset/tiled_dataset.py [hunk 3]
```diff
@@ -124,5 +125,30 @@ def shape(self):
...
+            axmin, axmax = ax.get_images()[0].get_clim()
+            vmin = axmin if axmin < vmin else vmin
+            vmax = axmax if axmax > vmax else vmax
+            ax.set_ylabel(" ")
+            ax.set_xlabel(" ")
+        if share_zscale:
+            for ax in fig.get_axes():
+                ax.get_images()[0].set_clim(vmin, vmax)
+        timestamp = self[0, 0].axis_world_coords("time")[-1].iso[slice_index]
+        fig.suptitle(f"TiledDataset {self.inventory['datasetId']} at time {timestamp} (slice={slice_index})", y=0.95)
```
Comment (Cadair) [inline dkist/dataset/tiled_dataset.py:151]: I think it would be better if this were "VBI Dataset (ASKJD) at time ..." if you don't mind.

**Pull request merged**